lib/events.js

```diff
@@ -365,40 +437,42 @@ EventEmitter.prototype.removeAllListeners =
     function removeAllListeners(type) {
       var listeners, events, i;
 
-      events = this._events;
+      events = this[kEvents];
       if (events === undefined)
         return this;
 
       // not listening for removeListener, no need to emit
-      if (events.removeListener === undefined) {
+      if (!events.has('removeListener')) {
         if (arguments.length === 0) {
-          this._events = Object.create(null);
+          this[kEvents] = new Map();
           this._eventsCount = 0;
-        } else if (events[type] !== undefined) {
-          if (--this._eventsCount === 0)
-            this._events = Object.create(null);
-          else
-            delete events[type];
+        } else if (events.delete(type)) {
+          --this._eventsCount;
         }
         return this;
       }
 
       // emit removeListener for all listeners on all events
       if (arguments.length === 0) {
-        var keys = Object.keys(events);
+        const len = events.size;
+        var keys = new Array(len);
+        i = 0;
+        for (var value of events.keys())
+          keys[i++] = value;
```

Comment:
Is this much faster than `var keys = [...events.keys()];`? I'm not sure that this would be particularly hot code

Reply:
https://bugs.chromium.org/p/v8/issues/detail?id=7973